enum.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true

+

+module Enumerable

+  def my_each

+    i = 0

+    while i < size

+      yield(self[i])

+      i += 1

+    end

+  	self

+  end

+

+  def my_each_with_index

+    i = 0

+    while i < size

+      yield(self[i], i)

+      i += 1

+    end

+  self

+  end

+

+  def my_select

+    i = 0

+    new_array = []

+    while i < size

+      new_array << self[i] if yield(self[i]) == true

+      i += 1

+    end

+    new_array

+  end

+

+  def my_all?

+    i = 0

+    while i < size

+      return false if yield(self[i]) == false || yield(self[i]).nil?

+

+      i += 1

+    end

+    true

+  end

+

+  def my_any?

+    i = 0

+    while i < size

+      return true if yield(self[i])

+

+      i += 1

+    end

+    false

+  end

+

+  def my_none?

+    i = 0

+    while i < size

+      return false if yield(self[i])

+      i += 1

+    end

+    true

+  end

+

+  def my_count


```

Comment:
My count is not working as expected you are getting a local jump error .use **return true if !block_given?** This should be used as a guard case in some of your methods.